### ISSUE TITLE
jq + wget shell one-liner

### ISSUE THREAD

**Issue by fluffy-critter** [nadimkobeissi/mkbsd#35]
```
curl https://storage.googleapis.com/panels-api/data/20240916/media-1a-i-p~s | jq '.data | .[] | select(.dhd) | .dhd' | xargs wget
```

**Comment by lilyvxv** [comment]
jq <3

**Comment by parkerlreed** [comment]
If you want you can have that go to a file and pass it in to `wget -i` so it will reuse the same connection as to not constantly disconnect and reconnect to the server.

**Comment by vs4vijay** [comment]
File is saved with wrong extension -`AdamM~Pixelized~Key3.jpeg@expires=1728508444399&fm=jpg&s=23cac714f045d42bd940bb17c58a2e0f%0D`

**Comment by fluffy-critter** [comment]
> File is saved with wrong extension -`AdamM~Pixelized~Key3.jpeg@expires=1728508444399&fm=jpg&s=23cac714f045d42bd940bb17c58a2e0f%0D`

Yeah there's probably some obscure wget option to make that work better, I've just been doing something like

```
for i in *@* ; do mv "$i" "$(echo $i | cut -f1 -d@)"
```

but the `-d` parameter varies by operating system (like on macOS and Linux it's `?`).

**Comment by lvxnull** [comment]
> Yeah there's probably some obscure wget option to make that work better
Not really unless you explicitly specify file names for every file

**Comment by lvxnull** [comment]
Not really a one-liner, but this will get rid of the weird file names
```sh
wget -qO- https://storage.googleapis.com/panels-api/data/20240916/media-1a-i-p~s | jq -r '.data[] | select(.dhd) | .dhd'  | while read -r url; do
  name="$(printf "%s" "$url" | awk -F/ '{print $NF}' | cut -d '?' -f 1)"
  wget "$url" -O "$name"
done
```

**Comment by parkerlreed** [comment]
`curl -OJ` automatically saves it as the content dispotion filename

```
wget -qO- https://storage.googleapis.com/panels-api/data/20240916/media-1a-i-p~s | jq -r '.data[] | select(.dhd) | .dhd' | while read line; do curl -OJ $line; done```

**Comment by fluffy-critter** [comment]
> `curl -OJ` automatically saves it as the content disposition filename, which strips the extra chars.
> 
> ```
> wget -qO- https://storage.googleapis.com/panels-api/data/20240916/media-1a-i-p~s | jq -r '.data[] | select(.dhd) | .dhd' | while read line; do curl -OJ $line; done
> ```

Well then in that case we should go full circle and make it a curl oneliner:

```
curl https://storage.googleapis.com/panels-api/data/20240916/media-1a-i-p~s | jq -r '.data[] | select(.dhd) | .dhd' | while read f ; do curl -OJ $f ; done
```
